### PR TITLE
Upgrade to git publish plugin for documentation

### PR DESCRIPTION
**Pull request by tgianos**
Due to a conflict with jgit and the Nebula plugins this work has been stuck in various branches of my personal fork for a long time. The latest versions of `nebula.netflixoss` have resolved some of those `jgit` conflicts and @sghill helped me resolve some other issues to hopefully get this working.

Once this is merged it should get rid of the final gradle related warning we have lying around during the build.

### CODE REVIEW

File: genie-ui/build.gradle [hunk 1]
```diff
@@ -1,6 +1,12 @@
-plugins {
-    id "com.moowork.node" version "1.2.0"
+buildscript {
+    repositories {
+        gradlePluginPortal()
```
Comment (mprimi) [inline genie-ui/build.gradle]: Isn't this already added globally in the main `build.gradle`?

File: build.gradle [hunk 2]
```diff
@@ -409,7 +410,7 @@ task collectDocumentation(group: "documentation", description: "Copy the documen
 }
 
 gitPublish {
-
+
```
Comment (mprimi) [inline build.gradle]: Adding some trailing whitespace here

Comment (tgianos) [reply build.gradle]: it was a copy/paste/modify from the developer example. Do you just mean it's an extra new line or that there are space characters somewhere?

File: genie-ui/build.gradle (hunk 1 above)
Comment (tgianos) [reply genie-ui/build.gradle]: @sghill is the one in the main `build.gradle` sufficient for these sub projects as well? You added it in both in your PR into my fork so I didn't touch it deferring to you as the expert but @mprimi might have a good point here

Comment (sghill) [reply genie-ui/build.gradle]: Good point @mprimi @tgianos. It should delegate to the repositories defined in the parent's build.gradle.

File: build.gradle (hunk 2 above)
Comment (mprimi) [reply build.gradle]: Yes it's a line with 4 spaces

**Pull request merged**